### PR TITLE
feat(core): apply account center field control

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Apply the account center control for Profile API and every field.

For better code review experience, the file `packages/core/src/routes/profile/index.ts` is not refactored for exeeding 300 lines, created LOG-10349 to do it later.

## Testing

Integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/profile/index.ts [hunk 1]
```diff
@@ -61,7 +71,20 @@ export default function profileRoutes<T extends UserRouter>(
...
+        'account_center.filed_not_editable'
+      );
+      assertThat(
+        avatar === undefined || fields.avatar === AccountCenterControlValue.Edit,
+        'account_center.filed_not_editable'
+      );
+      assertThat(
+        username === undefined || fields.username === AccountCenterControlValue.Edit,
+        'account_center.filed_not_editable'
+      );
```
Comment (simeng-li) [inline packages/core/src/routes/profile/index.ts:87]: use for each?

File: packages/core/src/routes/profile/middlewares/koa-interaction-sie.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import type { AccountCenter } from '@logto/schemas';
```
Comment (simeng-li) [inline packages/core/src/routes/profile/middlewares/koa-interaction-sie.ts:1]: The file name is not aligned with the middleware.

File: packages/core/src/routes/profile/middlewares/koa-interaction-sie.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import type { MiddlewareType } from 'koa';
+import { type IRouterParamContext } from 'koa-router';
+
+import type Queries from '#src/tenants/Queries.js';
+
+import assertThat from '../../../utils/assert-that.js';
```
Comment (simeng-li) [inline packages/core/src/routes/profile/middlewares/koa-interaction-sie.ts:7]:
```suggestion
import assertThat from '.#src/utils/assert-that.js';
```

File: packages/core/src/routes/profile/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/profile/index.ts:87]: I tried, but it is very hard to resolve typing issues. So I believe that won't be better

**Pull request merged**